src/app/services/topic.service.ts

```diff
@@ -1,9 +1,232 @@
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+      ],
+    },
+    {
+      name: 'JAVA',
+      subtopic: [
+        {
+          title: 'What is JAVA?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+        {
+          title: 'What is Lorem Ipsum?',
+          description:
+            'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+          content:
+            'Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.',
+        },
+      ],
+    },
+  ];
 
-  constructor() { }
+  constructor() {}
+
+  getListOfTopics(): Topic[] {
+    return this.TOPICSLIST;
+  }
+
+  getListOfSubtopicsByTopicName(topic: string): SubTopic[] | undefined {
+    return this.TOPICSLIST.filter((el) => el.name === topic)[0]?.subtopic;
```

Comment:
I dont really like this approach. If filter is empty you can return empty array. Undefined means that we dont have elements in the filtered array, but that means that we have an empty array. We will have less checks in the template if we use only arrays.